### PR TITLE
[Dask] Add example of using custom callback in Dask

### PR DESCRIPTION
**Pull request by hcho3**

### CODE REVIEW

**Review by trivialfis — COMMENTED** [review]
Thanks for the example.  Other than the comment inlined, I recommend using the `black` python formatter, it's awesome and its style is used by dask/distributed/optuna and many other projects.

File: demo/dask/callbacks.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)
+
+    dtrain = DaskDMatrix(client, X_train, y_train)
+    dtest = DaskDMatrix(client, X_test, y_test)
+
+    # Use train method from xgboost.dask instead of xgboost.  This
+    # distributed version of train returns a dictionary containing the
+    # resulting booster and evaluation history obtained from
+    # evaluation metrics.
+    with tempfile.TemporaryDirectory() as tmpdir:
```
[the remark this answers is not in the document]
Comment (hcho3) [reply demo/dask/callbacks.py:61]: Oops, let me remove it. I was testing the checkpoint callback.

File: demo/dask/callbacks.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
+"""Example of using callbacks in Dask"""
+import math
```
Comment (trivialfis) [inline demo/dask/callbacks.py:2]: Where is this used?

File: demo/dask/callbacks.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import numpy as np
+import xgboost as xgb
+from xgboost.dask import DaskDMatrix
+from dask.distributed import Client
+from dask.distributed import LocalCluster
+from dask import array as da
```
Comment (trivialfis) [inline demo/dask/callbacks.py:8]: Where is this used?

**Review by trivialfis — APPROVED** [review]
Left a few comments but looks good to me overall.  Thanks for the example!

**Pull request merged**